dump.go

```diff
@@ -123,11 +127,19 @@ func (data *Data) Dump() error {
 		data.MaxAllowedPacket = defaultMaxAllowedPacket
 	}
 
-	if err := meta.updateServerVersion(data.Connection); err != nil {
+	if err := data.getTemplates(); err != nil {
 		return err
 	}
 
-	if err := data.getTemplates(); err != nil {
+	// Start the read only transaction and defer the rollback until the end
+	// This way the database will have the exact state it did at the begining of
+	// the backup and nothing can be accidentally committed
+	if err := data.begin(); err != nil {
```

Comment:
We don't want to set single-transaction if the database is not purely InnoDb, which I assume we will pass that information by `LockTables`, so this `begin()` should be conditionalized.

Reply:
Actually the read only transaction makes no difference if it is on InnoDB or MyISAM. Because MyISAM is an in memory database this just doesn't have an affect for repeatable read. It does have the affect of protecting the database from the dump though so we don't need extra permissions and you can't accidentally commit anything.

Reply:
https://dev.mysql.com/doc/refman/8.0/en/commit.html

Reply:
These two settings are mutually exclusive, so we shouldn't assume this has no effect on MyISAM.